tests/server/api/routers/v1/test_datasets.py

```diff
@@ -315,3 +320,111 @@ async def test_get_dataset_download_specific_version(test_client, dataset_with_r
         )
     ).sort_index(axis=1)
     assert_frame_equal(actual, expected)
+
+
+async def test_post_dataset_upload_csv(test_client, session):
+    name = str(random())
```

Comment:
What does using randomness here buy us that a string would not?

Reply:
we wouldn't need to worry about name collision if it's a random string

Reply:
otherwise the test can be flaky if in the future we make the session fixture module-scoped

Reply:
NIt: A string such as `name = dataset-<name-of-test-case>` accomplishes the same thing without introducing randomness.

Reply:
That's good idea, but name-of-test-case can change. Since the name here is not in the critical path of this test, I think it suffices to use a random string without too much thought.

Reply:
OK I'll make it non-random.

Reply:
Missed this message. Sounds good. This one was a non-blocking suggestion.